### PR TITLE
Disable search button when search field is empty

### LINKED ISSUE #43: Disable search button when search field is empty
Currently, clicking the search button with an empty search field results a random page being loaded, which might be unexpected. It would be better to deactivate the search button whenever the search field is empty.

### PR DESCRIPTION
**Pull request by Jan-Ka**
➡ [**Live preview of the changes**](http://rawgit.com/waldyrious/primerpedia/disable-search-button/index.html)

This resolves #43 .

While implementing the required changes I thought about the submit call to `search` in the html. In theory this can be fired before the `window.onload` ran through. For that reason in `setSearchButtonEnabledState` i really wanted to ensure that we had the correct elements.

It does not really have any effect/impact on 99% of the executions.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
 		<div id='controls'>
 			<form onsubmit='search(); return false;'>
 				<input type='search' autocomplete='off' id='search-term' placeholder='Search...' autofocus='' />
-				<input type='submit' value='Search' />
+				<input id="searchButton" type='submit' value='Search' disabled="disabled"/>
```
Comment (waldyrious) [inline index.html]: I'd prefer not making changes in the formatting before the linting and refactoring PRs are merged, otherwise those changes will be all over the place, and the code will be in inconsistent state until then.

File: primerpedia.js [hunk 2]
```diff
@@ -38,10 +39,13 @@ function random() {
 }
 
 function search() {
+	setSearchButtonEnabledState();
```
Comment (waldyrious) [inline primerpedia.js]: I'd prefer calling this `updateSeachButtonState()` (or the more verbose `updateSeachButtonEnabledState()`), and reserve "setX" names for functions that actually receive the parameter to set (or that indicate in their name what the target will be set to).

File: primerpedia.js [hunk 3]
```diff
@@ -196,9 +200,33 @@ function getQueryVariable(parameter) {
...
+		searchTermInputElement instanceof HTMLInputElement;
+
+	var searchButtonAllowedTypes = ["submit", "button"];
+
+	var searchButtonValid = searchButton !== undefined &&
+		searchButton !== null &&
+		searchButton instanceof HTMLInputElement &&
+		searchButtonAllowedTypes.indexOf(searchButton.getAttribute("type")) > -1;
+
+	if(searchTermInputElementValid && searchButtonValid) {
```
Comment (waldyrious) [inline primerpedia.js]: While I comment the care to avoid errors, I think the toll to pay in this case is too great: the lines setting up protection against possible errors are more numerous than the actual code of the function!

Why wouldn't the simpler `if(searchTermInputElement && searchButton)` suffice?

Comment (Jan-Ka) [reply primerpedia.js]: On a second glance I also think it would suffice to just check if it's an instance of the appropriate type (HTMLElement / HTMLInpurtElement). As I said - in most cases this code will not be hit before `onload` ran.

I'm not ok with using code like `if(searchTermInputElement && searchButton)` ever (if these variables are not booleans) because it just ensures that these both variables are not falsy. However we require them to be of a specific value (or more precise - a specific instance).

File: primerpedia.js (hunk 2 above)
Comment (Jan-Ka) [reply primerpedia.js]: Makes sense

File: index.html (hunk 1 above)
Comment (Jan-Ka) [reply index.html]: I didn't think of that, thanks for pointing it out

File: primerpedia.js (hunk 3 above)
Comment (waldyrious) [reply primerpedia.js]: > I also think it would suffice to just check if it's an instance of the appropriate type (HTMLElement / HTMLInpurtElement).

Reducing the condition to a single test per element seems like a sensible middle-ground. It may not cover every single edge case, but I'd really prefer not to make the code too verbose without a strong justification, as every line added reduces its approachability and readability.

Please go ahead :)

**Comment by Jan-Ka** [comment]
Added the required changes as discussed, @waldyrious

**Comment by waldyrious** [comment]
I greatly appreciate the effort to make atomic commits. My preference (although I don't always manage to follow this myself) is to retain multiple commits in the final history of each branch when they represent atomic steps towards the overall change, rather than to reflect the history of the review process (which often includes some back and forth, as is the case with 6f3a223, for instance).

With this in mind, I'll squash the changes in this PR down to a single commit, but still merge it with `--no-ff` so that the graph properly indicates that there was indeed a PR and a review process. Hopefully this rationale will make sense to you.
